### PR TITLE
Consider config for NarrowReturnObjectTypeRule

### PR DESCRIPTION
**Pull request by mitelg**
See #12

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,5 @@
 composer.lock
 
 .phpunit.cache
+
+/.idea
```
Comment (samsonasik) [inline .gitignore]: please revert this, you should can ignore ide based config globally

https://sebastiandedeyne.com/setting-up-a-global-gitignore-file

Comment (mitelg) [reply .gitignore]: thanks for the link, was not aware of something like this :see_no_evil:

**Comment by mitelg** [comment]
as a run of the code style fixer fixed some unrelated files, I would assume, that https://github.com/rectorphp/type-perfect/blob/main/.github/workflows/code_analysis.yaml#L29 is not set up correctly :thinking: should this be `run: composer check-cs --ansi` instead?

**Comment by mitelg** [comment]
I adjusted the job, because it should have failed for example on this run https://github.com/rectorphp/type-perfect/actions/runs/9547786462/job/26313537179

**Review by samsonasik — APPROVED** [review]

**Comment by samsonasik** [comment]
Thank you @mitelg

**Pull request merged**